x-pack/solutions/observability/plugins/apm/server/routes/diagnostics/service_map/get_exit_spans_from_samples.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { termQuery, rangeQuery, termsQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  SPAN_DESTINATION_SERVICE_RESOURCE,
+  SPAN_ID,
+  PARENT_ID,
+  SERVICE_NAME,
+  SPAN_NAME,
+  TRACE_ID,
+} from '@kbn/apm-types';
+import type { APMEventClient } from '@kbn/apm-data-access-plugin/server';
+
+export async function getExitSpans({
+  apmEventClient,
+  start,
+  end,
+  destinationNode,
+  ids,
+}: {
+  apmEventClient: APMEventClient;
+  start: number;
+  end: number;
+  sourceNode: string;
+  destinationNode: string;
+  ids: string[];
+}) {
+  const response = await apmEventClient.search('diagnostics_get_exit_spans_from_source_node', {
+    apm: {
+      events: [ProcessorEvent.transaction],
+    },
+    track_total_hits: false,
+    size: 0,
+    query: {
+      bool: {
+        filter: [...rangeQuery(start, end), ...termsQuery(PARENT_ID, ...ids)],
+      },
+    },
+    aggs: {
+      matching_destination_resources: {
+        filter: {
+          term: {
+            [SERVICE_NAME]: destinationNode,
+          },
+        },
+        aggs: {
+          sample_docs: {
+            top_hits: {
+              size: 5,
+            },
+          },
+        },
+      },
+      destination_services: {
+        terms: {
+          field: SERVICE_NAME,
+          size: 50,
+        },
+        aggs: {
+          sample_docs: {
+            top_hits: {
+              size: 5,
+            },
```

Comment:
How about using `top_metrics` and specifying the fields we want to return? This would avoid returning the entire doc

Reply:
While `top_metrics` is more efficient and faster I used `top_metrics` on purpose  to get the entire doc. 


For diagnostic purposes, I believe having the full document is more valuable for analysis since the current UI might miss certain cases. Providing the complete documents allows users to download and share them with us, avoid the back and forth requests/

Unless there are any concerns about performance degradation, I'll keep `top_hits` wdyt?